platforms/ios/lib/WysiwygComposer/Tests/WysiwygComposerTests/Components/WysiwygComposerView/WysiwygComposerViewModelTests+MentionsState.swift

```diff
@@ -0,0 +1,127 @@
+        XCTAssertEqual(viewModel.getMentionsState(), result)
+        
+        viewModel.setMarkdownContent("[Room](https://matrix.to/#/!room:matrix.org)")
+        XCTAssertEqual(viewModel.getMentionsState(), result)
+    }
+    
+    func testMultipleMentionsBySettingThemIndividually() {
+        viewModel.setMention(url: "https://matrix.to/#/@alice:matrix.org", name: "Alice", mentionType: .user)
+        viewModel.setMention(url: "https://matrix.to/#/@bob:matrix.org", name: "Bob", mentionType: .user)
+        viewModel.setAtRoomMention()
+        
+        let mentionsState = viewModel.getMentionsState()
+        XCTAssertEqual(mentionsState.userIds.count, 2)
+        XCTAssertEqual(Set(mentionsState.userIds), ["@alice:matrix.org", "@bob:matrix.org"])
+        XCTAssertTrue(mentionsState.hasAtRoomMention)
+        XCTAssertTrue(mentionsState.roomIds.isEmpty)
+        XCTAssertTrue(mentionsState.roomAliases.isEmpty)
+    }
+    
+    func testMultipleDuplicateMentionsBySettingThemIndividually() {
+        viewModel.setMention(url: "https://matrix.to/#/@alice:matrix.org", name: "Alice", mentionType: .user)
+        viewModel.setMention(url: "https://matrix.to/#/@alice:matrix.org", name: "Alice", mentionType: .user)
+        
+        XCTAssertEqual(viewModel.getMentionsState(),
+                       MentionsState(userIds: ["@alice:matrix.org"], roomIds: [], roomAliases: [], hasAtRoomMention: false))
+    }
+    
+    func testMultipleMentionsBySettingThemWithContent() {
+        viewModel.setHtmlContent(
+            """
+            <p><a href=\"https://matrix.to/#/@alice:matrix.org\">Alice</a>, \
+            <a href=\"https://matrix.to/#/!room:matrix.org\">Room</a>, \
+            <a href=\"https://matrix.to/#/@bob:matrix.org\">Bob</a>, \
+            <a href=\"https://matrix.to/#/#room:matrix.org\">Room</a>, \
+            @room</p>
+            """
+        )
+        var mentionState = viewModel.getMentionsState()
+        XCTAssertEqual(Set(mentionState.userIds), ["@alice:matrix.org", "@bob:matrix.org"])
+        XCTAssertEqual(mentionState.roomAliases, ["#room:matrix.org"])
+        XCTAssertEqual(mentionState.roomIds, ["!room:matrix.org"])
+        XCTAssertTrue(mentionState.hasAtRoomMention)
+        
+        viewModel.setMarkdownContent(
+            """
+            [Room](https://matrix.to/#/!room:matrix.org), \
+            [Room](https://matrix.to/#/#room:matrix.org), \
+            [Alice](https://matrix.to/#/@alice:matrix.org), \
+            [Bob](https://matrix.to/#/@bob:matrix.org), \
+            @room
+            """
+        )
+        mentionState = viewModel.getMentionsState()
+        XCTAssertEqual(Set(mentionState.userIds), ["@alice:matrix.org", "@bob:matrix.org"])
+        XCTAssertEqual(mentionState.roomAliases, ["#room:matrix.org"])
+        XCTAssertEqual(mentionState.roomIds, ["!room:matrix.org"])
+        XCTAssertTrue(mentionState.hasAtRoomMention)
```

Comment:
This may be a separate test, otherwise it feels like we are testing two different things in the same function.